lib/phoenix/endpoint/attempt_code_reload_plug.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Phoenix.Endpoint.AttemptCodeReloadPlug do
```

Comment:
I wonder if we need this now?

Before we needed it because the handler would call the endpoint separately. But in this approach the handler no longer calls the endpoint, so it won't run into races.

In fact, the race we can run into is that we won't find the user handler (i.e. the UserSocket module), so we would need to wrap that instead.

Reply:
I guess we can try this:

1. Remove the protection from the old code, use your snippet above to reproduce the failure. Can you reproduce it?

2. If you can reproduce it, remove this protection and try again, can you reproduce it?

3. And then, try the snippet above changing `user_socket` and see if it happens in the new code

Reply:
We do. I can actually reproduce in both Cowboy and Bandit without this. It's the same issue, though it now occurs at the boundary b/w the underlying server and Phoenix (ie: Cowboy is looking to call `App.Endpoint.call/2`, but `App.Endpoint` doesn't exist since it's being recompiled). 

The intent of this change is to ensure that the module the web server is calling never goes away due to recompilation. `Cowboy2Handler` played that role previously.

Reply:
There's a similar race existing in the boundary between the underlying server and the socket implementation within Phoenix, but those aren't likely to get recompiled very often so it's less of a concern in practice.

Reply:
Perfect!